### ISSUE TITLE
List of mutable static state

### ISSUE THREAD

**Issue by sharwell** [gitextensions/gitextensions#4768]
:memo: This list is for my own use in implementing state reset functionality for the test suite. Current as of commit 8bf6ab9e80fb05753446d98d745dfda710c8432d.

* NBug
  * `BugReport.ProcessingException`
  * `ExceptionFilters.writeGeneratedAssemblyToDisk`
  * `InternalLogViewer.closed`
  * `InternalLogViewer.handleCreated`
  * `InternalLogViewer.initialized`
  * `InternalLogViewer.viewer`
  * `Logger.LogWritten`
  * `StoragePath.storagePathEnum`
  * `StoragePath.storagePathString`
  * `ZipStorer.CrcTable`
  * `SettingsOverride.Overridden`
  * *Many* fields and auto-properties in the `Settings` class
* AppVeyorIntegration
  * `AppVeyorAdapter.Projects`
  * `BuildDetails._buildDurationFormatter`
  * `JenkinsAdapter._buildDurationFormatter`
* GitStatistics
  * `Quadrilateral.QuadrilateralPointTypes`
  * `Quadrilateral.Empty`
* DeleteUnusedBranches
  * `SortableBranchesList.BranchesComparer.PropertyComparers`
* FindLargeFiles
  * `SortableObjectsList.GitObjectsComparer.PropertyComparers`
* Gerrit
  * `GerritPlugin._validatedHooks`
  * `GerritUtil.SshPathLocatorInstance`
* Github3
  * `GithubLoginInfo.Username`
  * `Github3Plugin.instance`
  * `Github3Plugin.github`
* GitUIPluginInterfaces
  * `ManagedExtensibility` class
* GitExtensionsVSIX
  * `GitExtCommands.Instance`
* GitPluginShared
  * `Commit.lastBranchCheck`
  * `Commit.lastFile`
  * `Commit._lastUpdatedCaption`
  * `PluginHelpers.AllowCaptionUpdate`
* GitCommands
  * `CommitTemplateManager.RegisteredTemplatesDic`
  * `ConfigFile.CommentChars`
  * `FileAssociatedIconProvider.LoadedFileIcons`
  * `FileHelper.BinaryExtensions`
  * `FileHelper.ImageExtensions`
  * `EnvironmentConfiguration.Env`
  * `GitCommandCache.CachedCommandsChanged`
  * `GitCommandCache.commandCache`
  * `GitCommandCache.queue`
  * `GitCommandHelpers.SshPathLocatorInstance`
  * `GitCommandHelpers._versionInUse`
  * `GitModule.GitDirectoryResolverInstance`
  * `GitModule._systemEncoding`
  * ~~`GitModule.DetachedPrefixes`~~
  * Many `GitVersion` fields
  * `PathUtil.EnvironmentAbstraction`
  * `PathUtil.EnvironmentPathsProvider`
  * `Settings.defaultInstance`
  * `Repositories._repositoryHistory`
  * `Repositories._remoteRepositoryHistory`
  * `Repositories._repositoryCategories`
  * `RevisionGraph.ShellGlobCharacters`
  * Many fields and properties in `AppSettings`
  * `StringExtensions.NewLineSeparator`
  * `WeakRefCache.Default`
  * `DetachedHeadParser.DetachedPrefixes`
* GitExtensions
  * `Settings.defaultInstance`
* GitExtUtils
  * `ControlThreadingExtensions._controlDisposed`
  * `ThreadHelper._joinableTaskContext`
  * `ThreadHelper._joinableTaskCollection`
  * `ThreadHelper._joinableTaskFactory`
* ResourceManager
  * `LocalizationHelpers.PlainCommitDataHeaderRenderer`
  * `Strings._instance`
  * `Translator._translation`
  * `Translator._name`
  * `TranslationUtl.UnTranslatableDLLs`
* GitUI
  * `CommitAutoCompleteProvider._regexes`
  * `FormContributors.tabCaptions`
  * `FormGitLog.instance`
  * `FormBrowse._showRevisionInfoNextToRevisionGrid`
  * `FormCreateTag.DropwdownTagOperation`
  * `FormGitIgnore.DefaultIgnorePatterns`
  * `SortableLostObjectsList.LostObjectsComparer.PropertyComparers`
  * `DiscussionHtmlCreator._systemInfoReplacement`
  * `DiscussionHtmlCreator._cssData`
  * `CombinedDiffHighlightService.Instance`
  * `DiffHighlightService.Instance`
  * `GitExtensionsForm.ApplicationIcon`
  * `GitExtensionsForm._windowPositionList`
  * `HotkeySettingsManager._serializer`
  * `HotkeySettingsManager.UsedKeys`
  * `MessageBoxes.instance`
  * `MouseWheelRedirector.instance`
  * `LoadedPlugins.Plugins`
  * `Settings.defaultInstance`
  * `RepoHosts.GitHosters`
  * `ScriptManager.Scripts`
  * `EditNetSpell._wordDictionary`
  * `FileStatusList._images`
  * `RevisionGrid.dashPattern`
  * `DvcsGraph.Junction.debugIdNext`
  * `MenuUtil.CaptionCustomMenuRenderer`
  * `MenuUtil.disabledFont`
  * `SingleHtmlUserManual._location`
  * `BuildInfoDrawingLogic._imageByBuildStatus`
  * `RepoObjectsTree.Node.DefaultContextMenus`

**Comment by RussKie** [comment]
FYI: I'm reworking `Repositories`  in #4766 and most statics will be gone when I'm done

**Comment by sharwell** [comment]
Yep it will be a while before I do anything about these. Also in the first round I wouldn't be doing a refactoring so much as just setting the values to defaults between tests.

**Comment by RussKie** [comment]
* `RepositoryHistoryManager.Locals`
* `RepositoryHistoryManager.Remotes`

**Comment by GintasS** [comment]
Abandoning this because of the age and lack of discussion.
